pkg/registry/rbac/clusterrole/policybased/storage.go

```diff
@@ -37,13 +37,15 @@ var groupResource = rbac.Resource("clusterroles")
 type Storage struct {
 	rest.StandardStorage
 
+	rest.StorageVersionProvider
+
 	authorizer authorizer.Authorizer
 
 	ruleResolver rbacregistryvalidation.AuthorizationRuleResolver
 }
 
-func NewStorage(s rest.StandardStorage, authorizer authorizer.Authorizer, ruleResolver rbacregistryvalidation.AuthorizationRuleResolver) *Storage {
-	return &Storage{s, authorizer, ruleResolver}
+func NewStorage(s rest.StandardStorage, svp rest.StorageVersionProvider, authorizer authorizer.Authorizer, ruleResolver rbacregistryvalidation.AuthorizationRuleResolver) *Storage {
```

Comment:
I could have simply added the StorageVersionProvider method to the StandardStorage interface, but I'm concerned about changing a published interface.

Reply:
You can test whether s implements the provider interface rather than make the callers all change, no?

Reply:
plus I'd rather have a nil svp here than an svp that panics if we call the Version() method :)

Reply:
Switched to type assertions.